### ISSUE TITLE
error on using type as generic alias

### ISSUE THREAD

**Issue by AllanDaemon** [microsoft/pyright#1130]
**Describe the bug**
When using `type` instead of `typing.Type` as PEP 585 proposes, it generates an error instead of working like `Type`

**To Reproduce**

Use this code with Pylancet:

```python
from typing import Type

class A: ...

t1: Type[A]
t2: type[A]
```

It generates the error: `Expected no type arguments`

**Expected behavior**

that `type[A]` works like `Type[A]`. Also, according to PEP 585, using `typing.Type` should raise a deprecation warning.

**VS Code extension or command-line**
Using VS Code with Python with vscode-pylance 2020.10.3 and Python 3.9 (Linux 64bits)

**Comment by erictraut** [comment]
Thanks for the bug report. I've updated the code to allow `type` to be used in place of `Type` for Python 3.9 and newer. 

I haven't added a deprecation warning. PEP 585 says that "type checkers may warn about such deprecated usage". These are going to be very common for some time, so deprecation warnings would introduce a lot of excess noise. Once Python 3.9 gets more adoption (perhaps once Python 3.10 has been released), we'll consider adding such a deprecation warning.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.83, which was just published. It will also be included in the next release of pylance.
